sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -745,6 +745,26 @@ object GpuDivModLike {
       }
     }
   }
+
+  /**
+   * Return a null row at "i" if either "dataCol[i]", "nullFlagCol1[i]" or "nullFlagCol2[i]"
+   * are null, otherwise return the row from the "dataCol".
+   */
+  def mergeNulls(
```

Comment:
nit: for me I think that the following would be slightly cheaper to compute. The probability that dataCol is larger than a boolean column is fairly high.  Also might be interesting to add in a check if any of nullFlag is true before doing the if/else

```
val nullFlag = withResource(nullFlagCol1.isNull) { nullFlag1 =>
  withResource(nullFlagCol2.isNull) { nullFlag2 =>
    nullFlag1.or(nullFlag2)
  }
}
withResource(nullFlag) { _ =>
  val needsToChange = withResource(nullFlag.any) { tmp =>
     tmp.isValid && tmp.getBoolean
  }
  if (needsToChange) {
    withResource(Scalar.fromNull(dataCol.getType)) { nullScalar =>
      nullFlag.ifElse(nullScalar, dataCol)
    }
  } else {
    dataCol.incRefCount()
  }
}
```